src/hotspot/share/opto/c2compiler.cpp

```diff
@@ -690,7 +690,7 @@ bool C2Compiler::is_intrinsic_supported(const methodHandle& method, bool is_virt
   case vmIntrinsics::_VectorUnaryOp:
   case vmIntrinsics::_VectorBinaryOp:
   case vmIntrinsics::_VectorTernaryOp:
-  case vmIntrinsics::_VectorBroadcastCoerced:
+  case vmIntrinsics::_VectorFromBitsCoerced:
```

Comment:
Where is Matcher::match_rule_supported() check?

Reply:
Currently all VectorSupport routines are in-lined during incremental lazy in-lining phase, this is to promote constant propagation post procedural in-lining during parsing, any intrinsic creation failure during lazy in-lining will result into emitting a call since at that point we do not have infrastructure (InlineTree) to perform procedure in-lining. 

But a match rule based check which is insensitive to constant ness of arguments could be done early, is it ok to take that as a separate RFE which cover all the VectorSupport.* routines. 

Also here we are piggybacking multiple operations over same C2 entry point i.e a vector broadcast,  maskall and fromLong hence it will not be possible to have a generic match rule supported check covering all the operations.

Reply:
The match_rule_supported check is in inline_vector_broadcast_coerced() as part of arch_supports_vector() checks.

Reply:
I see. Good.